stip/stip-001-burn-election.md

```diff
@@ -0,0 +1,127 @@
+# Abstract
+
+This proposal describes a mechanism for single-leader election using
+burn-mining. Leader election is intimately connected to the concept of
+chain selection-- as leaders must be able to choose their chain tip, and
+that choice is impacted by network participants' rules for chain selection.
+
+# Motivation
+
+Leader election on the Stacks chain enables:
+
+1. Near instantaneous confirmation of Stacks transactions
+2. High validation throughput
+3. An open leadership set
+4. Opportunity to participate without mining hardware
+5. Ability to migrate to a separate burn chain in the future
+
+# Election
+
+The basic structure for leader election through burn-mining is that
+for some Stacks block _N_, the leader is selected via some function of
+that leader's total cryptocurrency burnt in a block _N'_ on the
+underlying burn chain (or over a window of blocks).
+
+In such a system, if a candidate _Alice_ wishes to be a leader of a
+Stacks block, she issues a burn transaction in the underlying burn
+chain. The network then uses cryptographic sortition to choose a
+leader in a verifiably random process, weighted by burn amounts.
+The block in which this burn transaction is broadcasted is known
+as the "election block" for Stacks block _N_.
+
+## Committing to a chain tip
+
+Leaders must be able to choose which chain tip they want to build off
+of. Otherwise, malicious leaders would be able to force subsequent
+leaders to follow their chains, even if they misbehaved by hiding
+transactions, broadcasting partial blocks, etc.
+
+However, a healthy blockchain requires that leaders _commit_ to the
+chain tip they are working on at the time at which they expend energy,
+burn cryptocurrency, etc. This ensures that if a would-be malicious
+leader fails to win a block, they have suffered a penalty. For
+example, in Bitcoin, miners search for the inverse of a hash that
+contains the current chain tip. If they fail to win that block, they
+have wasted that energy. As they must attempt deeper and deeper forks
+to initiate a double-spend attacks, this becomes an exponential energy
+expenditure.
+
+It is important that this same property hold in the Stacks
+blockchain. The ability for leaders to engage in double-spend attacks
+at will is unacceptable. This property is enforced by making a leader
+announce their chain commitment _in the burn transaction_. However,
+unlike Bitcoin (and, for example, Bitcoin-NG) --- this commitment is a
+"lagging commitment". The chain tip announced in a burn transaction
+_cannot_ be the exact preceding chain tip, because a key advantage of
+using leader election is that a block may be _streamed_ -- namely, a
+leader is continuously confirming new transactions as long as their
+tenure lasts. So at a minimum, the commitment will be 1-block old.
```

Comment:
I was thinking the leader could do both.  They commit to a chain tip when they announce their burn in the sliding window, and the commit to a descendant of that chain tip when they start their tenure.  The former commitment stops the leader from committing to _any earlier microblock_ no matter how much they burn.  The latter commitment is used to decide (1) where in the microblock stream the new leader will begin its tenure on its fork, and (2) how much of a fork penalty to assign to the chain they built off of.

Reply:
True -- but even in this case, there is a period between when the leader is _known_ to be elected, and when they actually begin their block. The shorter that window, the less opportunity for malicious behavior.

Reply:
Thoughts on calling this a "lame-duck period"?